tests/test.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+generate_bats() {
+cat > "$1/test.bats" <<EOF
+#!/usr/bin/env bats
+
+load $TESTDIR/utils.sh
+
+@test "Testing $1" {
+    test_generate "$1"
+    test_compile "$1"
+    test_run "$1"
+}
+EOF
+}
+
+clean() {
+    rm -f packcc *.d/test.bats *.d/parser{,.c,.h}
+}
+
+main() {
+    set -e
+
+    export TESTDIR="$(cd "$(dirname "$0")" && pwd)"
+    export ROOTDIR="$TESTDIR/.."
+
+    cd "$TESTDIR"
+    clean
+
+    "${CC:-cc}" -o packcc $ROOTDIR/src/packcc.c
```

Comment:
Like CC, how about introducing PACKCC environment variable for specifying a packcc executable for testing?
Unless PACKCC is not specified, we can make a packcc executable here.

Reply:
Yes, good idea.